tree/ntuple/v7/src/RNTupleDescriptor.cxx

```diff
@@ -890,19 +907,42 @@ void ROOT::Experimental::RNTupleDescriptorBuilder::SetNTuple(
    fDescriptor.fGroupUuid = uuid;
 }
 
-void ROOT::Experimental::RNTupleDescriptorBuilder::AddField(
-   DescriptorId_t fieldId, const RNTupleVersion &fieldVersion, const RNTupleVersion &typeVersion,
-   std::string_view fieldName, std::string_view typeName, std::uint64_t nRepetitions, ENTupleStructure structure)
+ROOT::Experimental::RDanglingFieldDescriptor::RDanglingFieldDescriptor(
+   const RFieldDescriptor& fieldDesc) : fField(fieldDesc.Clone())
 {
-   RFieldDescriptor f;
-   f.fFieldId = fieldId;
-   f.fFieldVersion = fieldVersion;
-   f.fTypeVersion = typeVersion;
-   f.fFieldName = std::string(fieldName);
-   f.fTypeName = std::string(typeName);
-   f.fNRepetitions = nRepetitions;
-   f.fStructure = structure;
-   fDescriptor.fFieldDescriptors.emplace(fieldId, std::move(f));
+   fField.fParentId = kInvalidDescriptorId;
+   fField.fLinkIds = {};
+}
+
+ROOT::Experimental::RDanglingFieldDescriptor
+ROOT::Experimental::RDanglingFieldDescriptor::FromField(const Detail::RFieldBase& field) {
+   RDanglingFieldDescriptor fieldDesc;
+   fieldDesc.FieldVersion(field.GetFieldVersion())
+      .TypeVersion(field.GetTypeVersion())
+      .FieldName(field.GetName())
+      .TypeName(field.GetType())
+      .Structure(field.GetStructure())
+      .NRepetitions(field.GetNRepetitions());
+   return fieldDesc;
+}
+
+ROOT::Experimental::RResult<ROOT::Experimental::RFieldDescriptor>
+ROOT::Experimental::RDanglingFieldDescriptor::MakeDescriptor() const {
+   if (fField.GetId() == kInvalidDescriptorId) {
+      return R__FAIL("invalid field id");
+   }
+   if (fField.GetStructure() == ENTupleStructure::kInvalid) {
+      return R__FAIL("invalid field structure");
+   }
+   // FieldZero is usually named "" and would be a false positive here
+   if (fField.GetId() != DescriptorId_t(0)) {
+      Detail::RFieldBase::EnsureValidFieldName(fField.GetFieldName()).ThrowOnError();
```

Comment:
Should be passed with `R__FORWARD_RESULT`

Reply:
@jblomer I was having a little trouble forwarding here because the Result type mismatch
`outer: RResult<RFieldDescriptor>` vs `inner: RResult<void>` and wasn't really sure how to convert between the two properly.

Reply:
That's true.  Here's an attempt to address it with a new `R__FORWARD_ERROR` macro: #6039

Reply:
Awesome :) 👍

Reply:
@jblomer just rebased on #6039, thank you for the new feature :)